Solve issues of interrupting voice in process

paalex opened the pull request:
- if voice is in process, a new call would queue the voice requests
- to stop the voice, .stop() can be used as usual
- solved a bug with boolean callbacks not crossing the bridge to
  react-native. @[[NSNull null], @true] instead of @[@true] (which
  produces an unhandled promise.

paalex committed:
Solve issues of interrupting voice in process

- if voice is in process, a new call would queue the voice requests
- to stop the voice, .stop() can be used as usual
- solved a bug with boolean callbacks not crossing the bridge to
react-native. @[[NSNull null], @true] instead of @[@true] (which
produces an unhandled promise.

tarkanlar commented:
I am getting this error
`node_modules/react-native-speech/SpeechSynthesizer.m:45:5: Use of undeclared identifier 'utterance'`

paalex committed:
Fixed undeclared utterance

- fixed - SpeechSynthesizer.m:45:5: Use of undeclared identifier
'utterance'

paalex commented:
Sorry, fixed it now

bazzooka committed:
.gitignore is mandatory for such projects :)

bazzooka committed:
android sources for TTS

bazzooka committed:
choose the right OS version

bazzooka committed:
Maybe another sentence :) with promise test

bazzooka committed:
queue next message if forceStop is false or null

bazzooka committed:
now use a map of promise to manage multiple asynchronous resolve

bazzooka committed:
fix issue with android queue

bazzooka committed:
fix reject promise

bazzooka committed:
Remove multiple promise return

paalex commented:
Hey @tarkanlar ,
Can you check the latest commit?
It's working now.

tarkanlar commented:
Hey @paalex thanks, I just tried. I am getting this error 
![image](https://cloud.githubusercontent.com/assets/213978/20149468/17e74acc-a6ba-11e6-85d7-f4e11c4a92ce.png)

paalex committed:
Added time intervals, pitch, volume

also changed the error messages to Warn instead of Error

paalex committed:
Added event emitter for didFinishSpeechUtterance

paalex commented:
Hey @tarkanlar ,
I changed the error to Info only ( will show only in console log).
Also added control over time intervals before\after the utterance and control over pitch and volume.
Bonus - event emitter for didFinishSpeechUtterance event.

best

paalex committed:
Merge pull request #1 from bazzooka/master

Would like to merge the android support with updated ios library

tarkanlar commented:
Works good Thanks @paalex
